src/frontend/js/components/DashboardItem/DashboardItemCourseEnrolling.tsx

```diff
@@ -0,0 +1,288 @@
+    id: 'components.DashboardItemEnrollment.notEnrolled',
+    description: 'Text shown on a read-only not-enrolled target course',
+    defaultMessage: 'You are not enrolled in this course',
+  },
+  enrollCourse: {
+    id: 'components.DashboardItemEnrollment.enrollCourse',
+    description:
+      'Button shown on a read-only not-enrolled target course that redirects to the details view',
+    defaultMessage: 'Enroll',
+  },
+  statusClosed: {
+    id: 'components.DashboardItemEnrollment.statusClosed',
+    description: 'Text shown on a closed course run',
+    defaultMessage: 'CLOSED • Occurred on {startDate} - {endDate}',
+  },
+  statusOpened: {
+    id: 'components.DashboardItemEnrollment.statusOpened',
+    description: 'Text shown on a enrolled course run',
+    defaultMessage: 'OPEN • Happens on {startDate} - {endDate}',
+  },
+  statusNotActive: {
+    id: 'components.DashboardItemEnrollment.statusNotActive',
+    description: 'Text shown when a course run is not active',
+    defaultMessage: 'Not enrolled',
+  },
+  accessCourse: {
+    id: 'components.DashboardItemEnrollment.gotoCourse',
+    description: 'Button to access course when the user is enrolled',
+    defaultMessage: 'Access course',
+  },
+  runPeriod: {
+    id: 'components.DashboardItemEnrollment.runPeriod',
+    description: 'Text to display the period of a course run',
+    defaultMessage: 'On {startDate} - {endDate}',
+  },
+  enrolled: {
+    id: 'components.DashboardItemEnrollment.enrolled',
+    description: 'Text shown when user is enrolled in a course run',
+    defaultMessage: 'Enrolled',
+  },
+  enrollRun: {
+    id: 'components.DashboardItemEnrollment.enrollRun',
+    description: 'Button to enroll in a course run',
+    defaultMessage: 'Enroll',
+  },
+  enrollmentNotYetOpened: {
+    defaultMessage: 'Enrollment will open on {enrollment_start}',
+    description: 'Disclaimer that informs the user that enrollment is not yet opened',
+    id: 'components.DashboardItemCourseEnrollingRun.enrollmentNotYetOpened',
+  },
+  noCourseRunAvailable: {
+    defaultMessage: 'No session available for this course.',
+    description: 'Text displayed when no course run are opened for the course',
+    id: 'components.DashboardItemCourseEnrollingRun.noCourseRunAvailable',
+  },
+};
+
+interface Props {
+  course: AbstractCourse;
+  activeEnrollment?: Enrollment;
+  order?: Order;
+  writable?: boolean;
+  icon?: boolean;
+  notEnrolledUrl?: string;
+}
+
+export const DashboardItemCourseEnrolling = ({
+  course,
+  activeEnrollment,
+  writable,
+  order,
+  icon = false,
+  notEnrolledUrl = '#',
+}: Props) => {
+  return (
+    <div data-testid={'dashboard-item__course-enrolling__' + course.code}>
+      {!writable && (
+        <div className="dashboard-item__course-enrolling__infos">
+          {!!activeEnrollment && <Enrolled icon={icon} enrollment={activeEnrollment} />}
+          {!activeEnrollment && <NotEnrolled icon={icon} notEnrolledUrl={notEnrolledUrl} />}
```

Comment:
Better with a ternary nope ?

```suggestion
          {!activeEnrollment
            ? <NotEnrolled icon={icon} notEnrolledUrl={notEnrolledUrl} />            
            : <Enrolled icon={icon} enrollment={activeEnrollment} />
          }
```